core/src/main/java/com/flipkart/varadhi/core/cluster/ConsumerApi.java

```diff
@@ -1,17 +1,23 @@
 package com.flipkart.varadhi.core.cluster;
 
+import com.flipkart.varadhi.entities.GetMessagesRequest;
+import com.flipkart.varadhi.entities.Message;
 import com.flipkart.varadhi.entities.cluster.ConsumerInfo;
 import com.flipkart.varadhi.entities.cluster.ShardOperation;
 import com.flipkart.varadhi.entities.cluster.ShardStatus;
 
+import java.util.List;
 import java.util.concurrent.CompletableFuture;
 
 public interface ConsumerApi {
     CompletableFuture<Void> start(ShardOperation.StartData operation);
 
     CompletableFuture<Void> stop(ShardOperation.StopData operation);
 
+    CompletableFuture<Void> unsideline(ShardOperation.UnsidelineData operation);
```

Comment:
what does Future<> as the return value mean here? for unsideline op?
Given that the unsideline operation may take extended amount of time.